plan/physical_plan_builder.go

```diff
@@ -141,12 +141,8 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 	ts.allocator = p.allocator
 	ts.SetSchema(p.GetSchema())
 	ts.initIDAndContext(p.ctx)
-	txn, err := p.ctx.GetTxn(false)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if txn != nil {
-		ts.readOnly = txn.IsReadOnly()
+	if p.ctx.Txn() != nil {
+		ts.readOnly = p.ctx.Txn().IsReadOnly()
 	} else {
 		ts.readOnly = true
```

Comment:
Would `p.ctx.Txn()` be nil here? and why set `ts.readOnly = true` when `p.ctx.Txn()` is nil ?